internal/database/actions.go

```diff
@@ -12,203 +12,123 @@ import (
 	"github.com/upper/db/v4"
 )
 
 // Open returns a new database instance
-func Open(dbpath string) (*Database, error) {
+func Open(dbpath string) (*database, error) {
 	db, err := Connect(dbpath)
 	if err != nil {
 		return nil, err
 	}
-	return &Database{
+	return &database{
 		sess: db,
 	}, nil
 }
 
 // Database is a database instance to store measurements
-type Database struct {
+type database struct {
 	sess db.Session
 }
 
+var _ model.WritableDatabase = &database{}
+
 // Session returns the database session
-func (d *Database) Session() db.Session {
+func (d *database) Session() db.Session {
 	return d.sess
 }
 
-// ListMeasurements given a result ID
-func (d *Database) ListMeasurements(resultID int64) ([]MeasurementURLNetwork, error) {
-	measurements := []MeasurementURLNetwork{}
-	req := d.sess.SQL().Select(
-		db.Raw("networks.*"),
-		db.Raw("urls.*"),
-		db.Raw("measurements.*"),
-		db.Raw("results.*"),
-	).From("results").
-		Join("measurements").On("results.result_id = measurements.result_id").
-		Join("networks").On("results.network_id = networks.network_id").
-		LeftJoin("urls").On("urls.url_id = measurements.url_id").
-		OrderBy("measurements.measurement_start_time").
-		Where("results.result_id = ?", resultID)
-	if err := req.All(&measurements); err != nil {
-		log.Errorf("failed to run query %s: %v", req.String(), err)
-		return measurements, err
-	}
-	return measurements, nil
-}
-
-// GetMeasurementJSON returns a map[string]interface{} given a database and a measurementID
-func (d *Database) GetMeasurementJSON(measurementID int64) (map[string]interface{}, error) {
-	var (
-		measurement MeasurementURLNetwork
-		msmtJSON    map[string]interface{}
-	)
-	req := d.sess.SQL().Select(
-		db.Raw("urls.*"),
-		db.Raw("measurements.*"),
-	).From("measurements").
-		LeftJoin("urls").On("urls.url_id = measurements.url_id").
-		Where("measurements.measurement_id= ?", measurementID)
-	if err := req.One(&measurement); err != nil {
-		log.Errorf("failed to run query %s: %v", req.String(), err)
-		return nil, err
-	}
-	if measurement.Measurement.IsUploaded {
-		// TODO(bassosimone): this should be a function exposed by probe-engine
-		reportID := measurement.Measurement.ReportID.String
-		measurementURL := &url.URL{
-			Scheme: "https",
-			Host:   "api.ooni.io",
-			Path:   "/api/v1/raw_measurement",
-		}
-		query := url.Values{}
-		query.Add("report_id", reportID)
-		if measurement.URL.URL.Valid {
-			query.Add("input", measurement.URL.URL.String)
-		}
-		measurementURL.RawQuery = query.Encode()
-		log.Debugf("using %s", measurementURL.String())
-		resp, err := http.Get(measurementURL.String())
-		if err != nil {
-			log.Errorf("failed to fetch the measurement %s %s", reportID, measurement.URL.URL.String)
-			return nil, err
-		}
-		defer resp.Body.Close()
-		if err := json.NewDecoder(resp.Body).Decode(&msmtJSON); err != nil {
-			log.Error("failed to unmarshal the measurement_json")
-			return nil, err
-		}
-		return msmtJSON, nil
-	}
```

Comment:
I understand that here your intent was to group related functions together. It makes sense to do that but it does not make sense to do that _now_. By reordering the functions you are making what would be a very obvious and short diff much more complex to read and validate. Please, restore the previous order of the function such that the diff is minimal and obvious.